controllers/datasource_controller.go

```diff
@@ -327,6 +334,7 @@ func (r *GrafanaDatasourceReconciler) onDatasourceCreated(ctx context.Context, g
 
 func (r *GrafanaDatasourceReconciler) UpdateStatus(ctx context.Context, cr *v1beta1.GrafanaDatasource) error {
 	cr.Status.Hash = cr.Hash()
+	cr.Status.NoMatchingInstances = true
```

Comment:
Could you, please, explain why this line is added here and not inside a function that checks for presence of matching instances? (assuming `UpdateStatus` can be called from anywhere)
Btw, I see that in different controllers it is set to different values.